test/parallel/test-node-run.js

```diff
@@ -222,4 +222,46 @@ describe('node --run [command]', () => {
     assert.strictEqual(child.stdout, '');
     assert.strictEqual(child.code, 1);
   });
+
+  it('runs script in a custom working directory using --run-dir', async () => {
+    const customDir = fixtures.path('run-script');
+    const child = await common.spawnPromisified(
+      process.execPath,
+      [ '--run-dir', customDir, '--run', `pwd${envSuffix}` ],
+
+      { cwd: fixtures.path('run-script/sub-directory') }
+    );
+
+    assert.strictEqual(child.stdout.trim(), customDir);
+    assert.strictEqual(child.stderr, '');
+    assert.strictEqual(child.code, 0);
+  });
+
+  it('returns error on non-directory path for --run-dir', async () => {
+    const nonDirPath = fixtures.path('run-script/package.json');
+
+    const child = await common.spawnPromisified(
+      process.execPath,
+      [ '--run-dir', nonDirPath, '--run', 'pwd' ],
+      { cwd: fixtures.path('run-script') }
+    );
+
+    assert.match(child.stderr, /not a directory/);
+    assert.strictEqual(child.stdout, '');
+    assert.strictEqual(child.code, 1);
+  });
+
+  it('returns error when --run-in is used without --run', async () => {
+    const dirPath = fixtures.path('run-script/package.json');
+
+    const child = await common.spawnPromisified(
+      process.execPath,
+      [ '--run-in', dirPath ],
```

Comment:
```suggestion
  it('returns error when --run-dir is used without --run', async () => {
    const dirPath = fixtures.path('run-script/package.json');

    const child = await common.spawnPromisified(
      process.execPath,
      [ '--run-dir', dirPath ],
```

Reply:
thanks I thought I should test with the wrong command, I thought it was to catch an error